build/gtk-bundle-from-msys2.sh

```diff
@@ -0,0 +1,289 @@
+libgpg-error
+libgit2
+libidn
+libjpeg-turbo
+libogg
+libsoup
+libssh2
+libsystre
+libtasn1
+libtheora
+libtiff
+libtre-git
+libvorbis
+libvorbisidec-svn
+libwebp
+libxml2
+libxslt
+lua51
+nettle
+openssl
+orc
+p11-kit
+readline
+rtmpdump-git
+sqlite3
+termcap
+xz
+"
+
+gtk2_dependency_pkgs="
+webkitgtk2
+"
+gtk3_dependency_pkgs="
+webkitgtk3
+"
+
+package_urls=""
+
+
+handle_command_line_options() {
+	for opt in "$@"; do
+		case "$opt" in
+		"-c"|"--cache")
+			use_cache="yes"
+			;;
+		"-z"|"--zip")
+			make_zip="yes"
+			;;
+		"-2")
+			gtkv="2"
+			;;
+		"-3")
+			gtkv="3"
+			;;
+		"-h"|"--help")
+			echo "gtk-bundle-from-msys2.sh [-c] [-h] [-z] [-2 | -3] [CACHEDIR]"
+			echo "      -c Use pacman cache. Otherwise pacman will download"
+			echo "         archive files"
+			echo "      -h Show this help screen"
+			echo "      -z Create a zip afterwards"
+			echo "      -2 Prefer gtk2"
+			echo "      -3 Prefer gtk3"
+			echo "CACHEDIR Directory where to look for cached packages (default: /var/cache/pacman/pkg)"
+			exit 1
+			;;
+		*)
+			cachedir="$opt"
+			;;
+		esac
+	done
+}
+
+initialize() {
+	if [ -z "$cachedir" ]; then
+		cachedir="/var/cache/pacman/pkg"
+	fi
+
+	if [ "$use_cache" = "yes" ] && ! [ -d "$cachedir" ]; then
+		echo "Cache dir \"$cachedir\" not a directory"
+		exit 1
+	fi
+
+	gtk="gtk$gtkv"
+	eval "gtk_dependency_pkgs=\${${gtk}_dependency_pkgs}"
+
+	pkgs="
+${packages}
+${gtk_dependency_pkgs}
+"
+}
+
+_remember_package_source() {
+	if [ "$use_cache" = "yes" ]; then
+		package_url=`pacman -Sp mingw-w64-${ABI}-${2}`
+	else
+		package_url="${1}"
+	fi
+	package_urls="${package_urls}\n${package_url}"
+}
+
+_getpkg() {
+	if [ "$use_cache" = "yes" ]; then
+		package_info=`pacman -Qi mingw-w64-$ABI-$1`
+		package_version=`echo "$package_info" | grep "^Version " | cut -d':' -f 2 | tr -d '[[:space:]]'`
+		ls $cachedir/mingw-w64-${ABI}-${1}-${package_version}-* | sort -V | tail -n 1
+	else
+		pacman -Sp mingw-w64-${ABI}-${1}
+	fi
+}
+
+extract_packages() {
+	# hack for libxml2 postinstall script which expects "bin/mkdir"
+	mkdir -p bin
+	cp /bin/mkdir bin/
+	# extract packages
+	for i in $pkgs; do
+		pkg=$(_getpkg $i)
+		_remember_package_source $pkg $i
+		if [ "$use_cache" = "yes" ]; then
+			if [ -e "$pkg" ]; then
+				echo "Extracting $pkg from cache"
+				tar xaf $pkg
+			else
+				echo "ERROR: File $pkg not found"
+				exit 1
+			fi
+		else
+			echo "Download $pkg using curl"
+			curl -L "$pkg" | tar -x --xz
+		fi
+		if [ -f .INSTALL ]; then
+			echo "Running post_install script for \"$i\""
+			/bin/bash -c ". .INSTALL; post_install"
+		fi
+		rm -f .INSTALL .MTREE .PKGINFO .BUILDINFO
+	done
+}
+
+move_extracted_files() {
+	echo "Move extracted data to destination directory"
+	if [ -d mingw32 ]; then
+		for d in bin etc home include lib libexec locale sbin share ssl var; do
+			if [ -d "mingw32/$d" ]; then
+				rm -rf $d
+				# prevent sporadic 'permission denied' errors on my system, not sure why they happen
+				sleep 0.5
```

Comment:
As discussed on IRC, it *does* delete only $d which is intended here because mingw32/$d is moved to $d afterwards.